### PR TITLE
Adding missing params

### PR DESCRIPTION
**Pull request by GimmeYerGold**
@kentongray Lemme know if I missed anything. From here I'll move ahead with displaying better request examples.

### CODE REVIEW

File: api/endpoints/reservation.md [hunk 1]
```diff
@@ -57,6 +57,10 @@ Collection of all Reservations.
     + includeNonCancelled = `true` (optional, boolean, , `false`) ... If true, query will include non-cancelled reservations
     + includePastReservations = `false` (optional, boolean, `true`) ... If true, query will include past reservations
     + showOnlyMyReservations = `true` (optional, boolean, `false`) ... If true, query will only include logged in user's reservations
+    + startDate (optional, number, `1404410211910`) ... Epoch time of reservation start date
+    + endDate (optional, number, `1404421051661`) ... Epoch time of reservation end date
+    + buildingId (optional, number, `44`) ... Id of the building to query from
+    + roomId (optional, number, `16`) ... Id of the room to query from
```
Comment (rmoscowitz) [inline api/endpoints/reservation.md]: Can you double check these? I think these are `building`/`room`, not `buildingId`/`roomId`.

File: api/endpoints/room.md [hunk 2]
```diff
@@ -73,11 +73,15 @@ Collection of all Rooms.
...
+    + includeOccupiable (optional, boolean, `true`) ... If true, query will include occupiable rooms
+    + includeNonOccupiable (optional, boolean, `false`) ... If true, query will include non-occupiable rooms
     + locationSearch (optional, string, `houston tx`) ... Will do a location lookup based on the search string.
     + nearLatitude (optional, number, `29.75613`) ... Latitude to base search on (50 mile radius)
     + nearLongitude (optional, number, `-95.39425`) ... Longitude to base search on (50 mile radius)
     + startDate (optional, number, `1404410211910`) ... Epoch time of room availablilty start date
     + endDate (optional, number, `1404421051661`) ... Epoch time of room availablilty end date
+    + numberOfPeople (optional, number, `6`) ... Number to search for room by maximum occupancy
+    + includeParking (optional, boolean, `true`) ... If true, query will include rooms with parking spaces
     + search (optional, string, `conference`) ... String to search for room by name
```
Comment (rmoscowitz) [inline api/endpoints/room.md]: `search` is in the list of global options; should it also be here?

File: api/endpoints/room.md [hunk 3]
```diff
@@ -73,11 +73,15 @@ Collection of all Rooms.
 + Parameters
```
Comment (rmoscowitz) [inline api/endpoints/room.md:73]: Should the `## Room Collection...` line above be updated with the new params too?

File: api/endpoints/user.md [hunk 4]
```diff
@@ -100,7 +100,9 @@ Collection of all Users.
     + search (optional, string, `mario`) ... String to search for user by firstName, lastName, or email
```
Comment (rmoscowitz) [inline api/endpoints/user.md]: `search` is in the list of global options; should it also be here?

File: api/endpoints/user.md [hunk 5]
```diff
@@ -100,7 +100,9 @@ Collection of all Users.
...
     + centerId (optional, number, `12`) ... Id of center to query from. If `centerId` is provided `role` is _required_
     + role (optional, string, `operator`) ... Role name to query from. Valid values are: _administrator_, _operator_, _customer_, _manager_, and _technician_. If `role` is provided `centerId` is _required_
-
```
Comment (rmoscowitz) [inline api/endpoints/user.md:103]: Also `usertypeid` (int)

File: api/endpoints/user.md [hunk 6]
```diff
@@ -100,7 +100,9 @@ Collection of all Users.
     + search (optional, string, `mario`) ... String to search for user by firstName, lastName, or email
     + centerId (optional, number, `12`) ... Id of center to query from. If `centerId` is provided `role` is _required_
     + role (optional, string, `operator`) ... Role name to query from. Valid values are: _administrator_, _operator_, _customer_, _manager_, and _technician_. If `role` is provided `centerId` is _required_
-    
+    + siteAdmin (optional, boolean, `true`) ... If true, query will include users with Admin permissions
+    + hasRoom (optional, boolean, `true`) ... If true, query will include users with rooms
+    + hasDefaultVisitorCenter (optional, boolean, `true`) ... If true, query will include users with a default visitor center
```
Comment (rmoscowitz) [inline api/endpoints/user.md:171]: The last section of this file... I'm not sure what that is...

```
 ### Hold a Request [GET]
 + Response 200
     [User][]
```

It also looks like there are some things in `UserResource.scala` that aren't in this file yet:

```
/me/reportAuth
/me/supportAuth
/{id}/resetPassword
/{id}/resetPassword
/{id}/permissions GET & PUT
{id}/attachAvatar
username/{username}
```

**Review by rmoscowitz — CHANGES_REQUESTED** [review]

**Comment by GimmeYerGold** [comment]
@rmoscowitz okay, I think I addressed all your comments, but let me know if there are any other edits needed. Thanks!

**Comment by GimmeYerGold** [comment]
@rmoscowitz @kentongray added one more edit per Nate's request: change 'Remove a Request' to 'Cancel a Request' for more customer clarity

**Comment by jmlopez-rod** [comment]
Don't forget to generate the html file.

File: api/endpoints/user.md [hunk 7]
```diff
@@ -97,10 +140,12 @@ Collection of all Users.
 ### List of all Users [GET]
 
 + Parameters
-    + search (optional, string, `mario`) ... String to search for user by firstName, lastName, or email
     + centerId (optional, number, `12`) ... Id of center to query from. If `centerId` is provided `role` is _required_
+    + userTypeId (optional, number, `23`) ... If of user type to query from.
```
Comment (jmlopez-rod) [inline api/endpoints/user.md]: `If`? you mean `Id`?

**Comment by GimmeYerGold** [comment]
@jmlopez-rod okay, just did a commit with the index file. Thanks!

File: api/endpoints/user.md [hunk 8]
```diff
@@ -122,10 +167,165 @@ The following attributes are required to create a User: `firstName`, `lastName`,
...
+
+## Get User Permission Data [/users/{{id}/permissions]
+
+Get user permission data
+
+### Retrieve User Permissions [GET]
+
++ Response 200 (application/json)
+
+            [
```
Comment (jmlopez-rod) [inline api/endpoints/user.md:258]: Can you reformat this json block? https://jsonformatter.curiousconcept.com/

**Review by jmlopez-rod — CHANGES_REQUESTED** [review]

**Comment by GimmeYerGold** [comment]
@jmlopez-rod next round of edits done !

**Review by jmlopez-rod — APPROVED** [review]

File: api/endpoints/user.md [hunk 9]
```diff
@@ -122,10 +167,165 @@ The following attributes are required to create a User: `firstName`, `lastName`,
...
+
+### Retrieve Current User Report Auth Data [GET]
+
++ Response 200 (application/json)
+
+            {
+                "server:" "http://andross.corp.iofficecorp.com",
+                "authToken:" "a234893fmdfjhdkjh345897djhdjkh59",
+                "orgId:" "dev_fmcloud",
+                "username: "foxmccloud"
```
Comment (rmoscowitz) [inline api/endpoints/user.md:226]: Can we use fake info here?

File: api/endpoints/user.md [hunk 10]
```diff
@@ -122,10 +167,165 @@ The following attributes are required to create a User: `firstName`, `lastName`,
...
+
+Get current user report auth data
+
+### Retrieve Current User Support Auth Data [GET]
+
++ Response 200 (application/json)
+
+            {
+                "server:" "http://support.iofficecorp.com",
+                "authToken:" "imp_46436454_3452345345dgdg3q5345dsgdgsdg_=dev_fmccloud_420"
```
Comment (rmoscowitz) [inline api/endpoints/user.md:239]: Same question here, can this be fake data instead?

File: api/endpoints/user.md (hunk 8 above)
Comment (rmoscowitz) [reply api/endpoints/user.md:258]: Also, can we change some of the roles to show examples of the other roles?

File: api/endpoints/user.md (hunk 6 above)
Comment (rmoscowitz) [reply api/endpoints/user.md:171]: Two still missing, I think:

```
/{id}/permissions PUT
{id}/attachAvatar
```

File: api/endpoints/user.md (hunk 9 above)
Comment (GimmeYerGold) [reply api/endpoints/user.md:226]: @rmoscowitz lol i thought this was fake 😄 I just typed a bunch of random stuff for authToken, and followed the Starfox references for the other stuff, including the server name.

**Comment by GimmeYerGold** [comment]
@rmoscowitz okee dokee, i think I got the rest of your comments on here :)

**Review by rmoscowitz — APPROVED** [review]

**Pull request merged**